features/step_definitions/kata.rb

```diff
@@ -1,229 +1,127 @@
   step %Q|I obtain test data file "kata/catalogsource.yaml"|
-  user.cli_exec(:apply, f: 'catalogsource.yaml')
-  step %Q/a pod becomes ready with labels:/, table(%{
-    | olm.catalogSource=redhat-marketplace |
-  })
+  @result_catalog = user.cli_exec(:apply, f: "catalogsource.yaml")
+  raise "Failed to install sandboxed-operator" unless @result_catalog[:success]
+  step %Q|I obtain test data file "kata/subscription.yaml"|
+  @result_subcription = user.cli_exec(:apply, f: "subscription.yaml")
+  raise "Failed to install sandboxed-operator" unless @result_subcription[:success]
 end
 
-Given /^the kata-operator is installed(?: to #{OPT_QUOTED})? using OLM(?: (CLI|GUI))?$/ do | kata_ns, install_method |
-  ensure_admin_tagged
-  kata_ns ||= "openshift-sandboxed-containers-operator"
-  kata_config_name = "example-kataconfig"
-  step %Q/I store master major version in the :master_version clipboard/
-  raise "Kata operator OLM installation only supported for OCP >= 4.8" unless cb.master_version >= "4.8"
-  install_method ||= 'CLI'
-  # first check pre-req
-  step %Q/I switch to cluster admin pseudo user/
-  project('openshift-marketplace')
-  unless catalog_source('qe-app-registry').exists?
-    logger.info("Kata installation depends on `qe-app-registry`, which is missing in this cluster, calling step to create it...")
-    step %Q/I create "qe-app-registry" catalogsource for testing/
-  end
+Then /^sandboxed-operator operator should be installed and running$/ do
+  step %Q/I wait until sandboxed operator is ready/
+end
 
-  unless kata_config(kata_config_name).exists?
-    if install_method == 'GUI'
-      package_name = 'sandboxed-containers-operator'
-      catalog_name = 'qe-app-registry'
-      @result = admin.cli_exec(:create_namespace, name: kata_ns)
-      project(kata_ns)
-      step %Q/I switch to the first user/
-      step %Q/the first user is cluster-admin/
-      step %Q(I use the "#{kata_ns}" project)
-      step %Q/I open admin console in a browser/
-      step %Q/the step should succeed/
-      step %Q/I perform the :goto_operator_subscription_page web action with:/, table(%{
-        | package_name     | #{package_name} |
-        | catalog_name     | #{catalog_name} |
-        | target_namespace | #{kata_ns}      |
-      })
-      step %Q/the step should succeed/
-      # save the channel from subscription into cb.channel
-      step %Q/I extract the channel information from subscription and save it to the clipboard/
-      step %Q/I perform the :set_custom_channel_and_subscribe web action with:/, table(%{
-        | update_channel    | #{cb.channel} |
-        | install_mode      | OwnNamespace  |
-        | approval_strategy | Automatic     |
-      })
-      step %Q/the step should succeed/
-      step %Q/a pod becomes ready with labels:/, table(%{
-        | control-plane=controller-manager |
-      })
+Given /^I wait until sandboxed operator is ready$/ do
```

Comment:
@valtair1891 we already have the step `a pod becomes ready with labels`, so I'm curious why you chose to write a kata operator specific step.  I think you can use that instead.  Also, are you sure the index of the the `.status.components.refs` are always the same?  Checking a pod with label is more reliable.